docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,158 @@
+
+In a future release of EdgeX, there is a desire to allow 3rd party systems to make requests of the southside via message bus.  Specifically, a 3rd party system will send a command request via message to the command service via the allowed message bus implementations (which could be MQTT or Redis Pub/Sub today).  The command service would then relay the message request via message bus to the managing device service.  (The command service would also perform any translations on the request as it does for REST requests today.) The device service would use the message to trigger action on the device/sensor as it does when it receives a REST request today and respond via message bus back to the command service.  In turn, the command service would respond to the 3rd party system via message bus.
+
+### Design
+
+The command service requires the means to subscribe to the EdgeX or external (as a 3rd party may want to use their message bus to send commands) message bus.  It would use the messaging client (go-mod-messaging) to create a new MessageClient, connect to a message bus, and subscribe to a configured message topic.
+
+The command service would also publish messages to device service topics using the go-mod-messaging MessageClient.  It would relay any command request to a designated topic for the device service which would be subscribed to the topic.
+
+Likewise, the device services must use the messaging client (go-mod-messaging for Go services and C alternative for C services) to create a new MessageClient, connect to the message bus and subscribe to configured message topics to receive actuation messages from the command service.
+
+Because any actuation request may require a response on the part of EdgeX, the command service must provide the means to receive an optional response message from the device services and publish a response back to the 3rd party caller via message bus as well.
+
+The command message request messages should contain a correlation identifier which would be added to any response so that the 3rd party system would know to associate the responses to the original request.
+
+The device service message command subscription, response publishing and general handling should be implemented in the SDKs versus individually in the device services.
+
+!!! Note
+    Message responses are not guaranteed (on the part of the device service to the command service or from the command service back to the originating command requester).  Both the command service and the original 3rd party system making the command request bear the burden to manage timeouts (and possibly resending the request) if there is no response.
```

Comment:
How would this work? Would the command service define a single response timeout setting, or should this be settable on a per-device service, or possibly even on a per-request basis?

If core command times out waiting for a response, would it send its own response indicating a timeout occurred?

Reply:
> If core command times out waiting for a response, would it send its own response indicating a timeout occurred?

IMO, yes send it back to 3rd party requester.

Reply:
per lenny above.  Added extra comment in this section of the doc.